kardinal/reconciler/reconciler.go

```diff
@@ -95,3 +113,56 @@ func Reconcile(ctx context.Context, cl client.Client) error {
 
 	return nil
 }
+
+// OPERATOR-TODO make sure to execute this again once we connect the operator to listen to k8s Deployments and Services events
+// OPERATOR-TODO there is another approach we could take, if it doesn't works for all use cases, which is to use MutatingAdmissionWebHooks
+// related info for this here: https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation and particularly this https://book.kubebuilder.io/reference/webhook-for-core-types
+// for creating and webhook for these core types.
+func reconcileIstioLabelsForBaselineServicesAndDeployments(ctx context.Context, cl client.Client, clusterResources *resources.Resources) error {
```

Comment:
Can we move this logic under the `resources` package and make it a function of `Resources`.  Can we also update the labels only if there are changes similar to what we do for other updates.

Reply:
good suggestion!
done! and I also added some tests for the Istio labels ensurer